src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -78,6 +75,23 @@ public HttpMethod(string method)
             _method = method;
         }
 
+        // Private constructor used internally to create an HttpMethod with a pre-calculated
+        // hash code. The method parameter must already be uppercase, which allows us to use
+        // method.GetHashCode() instead of method.ToUpperInvariant().GetHashCode(), avoiding
+        // the unnecessary extra string allocation. The ignored parameter is just there to
+        // distinguish this constructor overload from the public constructor.
+        private HttpMethod(string method, bool ignored)
```

Comment:
Done. (I originally had it chaining the way you suggest, but changed it as an extreme micro-optimization, that doesn't matter much anyway).